src/main/java/ee/carlrobert/codegpt/toolwindow/chat/ChatToolWindowTabPanelEditor.java

```diff
@@ -102,6 +122,25 @@ private JPanel createHeaderComponent(String language) {
 
   private JPanel createHeaderActions() {
     var wrapper = new JPanel(new FlowLayout(FlowLayout.LEFT, 0, 0));
+    wrapper.add(new IconActionButton("Diff", AllIcons.Actions.DiffWithClipboard, new AnAction() {
+      @Override
+      public void actionPerformed(@NotNull AnActionEvent event) {
+          String text = editor.getDocument().getText();
+          Application application = ApplicationManager.getApplication();
+          var locationOnScreen = ((MouseEvent) event.getInputEvent()).getLocationOnScreen();
+          locationOnScreen.y = locationOnScreen.y - 16;
+          application.invokeLater(() -> application.runWriteAction(() -> WriteCommandAction.runWriteCommandAction(event.getProject(), () -> {
+              try {
+                  assert event.getProject() != null;
+                  Editor editor = FileEditorManager.getInstance(event.getProject()).getSelectedTextEditor();
+                  openDiffView(event.getProject(), editor, text);
+                  OverlayUtils.showInfoBalloon("Diff opened!", locationOnScreen);
+              } catch (Exception ignore) {
+                  OverlayUtils.showInfoBalloon("Diff failed!", locationOnScreen);
+              }
+          })));
+      }
+    }));
     wrapper.add(new IconActionButton("Copy", AllIcons.Actions.Copy, new AnAction() {
```

Comment:
Let's add a gap before the Copy button.
`wrapper.add(Box.createHorizontalStrut(8));`